thrift-logger/src/main/java/com/pinterest/singer/client/logback/AuditableLogbackThriftLoggerFactory.java

```diff
@@ -36,20 +49,86 @@
 
   private static Logger LOG = LoggerFactory.getLogger(AuditableLogbackThriftLoggerFactory.class);
 
+  private static final String AUDIT_THRIFT_LOGGER_CLIENT_INIT_EXCEPTION = "audit.thrift_logger.audit_client.init.exception";
+  private static LoggingAuditClient loggingAuditClient;
+
   private final ContextBase contextBase = new ContextBase();
+  private final File basePath;
+  private final int rotateThresholdKBytes;
+
 
+  @Deprecated
+  public AuditableLogbackThriftLoggerFactory(File basePath, int rotateThresholdKBytes) {
+    this.basePath = basePath;
+    this.rotateThresholdKBytes = rotateThresholdKBytes;
+  }
+
+  public AuditableLogbackThriftLoggerFactory() {
+    basePath = null;
+    rotateThresholdKBytes = -1;
+  }
+
+  public static void createLoggingAuditClientFromConfigFile(String filePath){
```

Comment:
Seems like an unused method

Reply:
Why are we using a static variable inside the factory? Shouldn't the factory itself be a singleton rather than the client?

Reply:
(1) the static method will be used in applications that uses ThriftLogger and can have audit configuration specified in a file.  This is just another way to create LoggingAuditClient and will help in migration from existing LogbackThriftLoggerFactory to AuditableLogbackThriftLoggerFactory.

(2) As a static variable of the factory, LoggingAuditClient is shared by all AuditableLogbackThriftLoggers created using the factory.  The factory itself is a singleton according the ThriftLoggerFactory.java.

Reply:
Still recommend we should use instance variables and static reference in SingerSettings etc.

Reply:
I feel it is easier to make it static in AuditableLogbackThriftLoggerFactory.  The actual logger can easily reference the audit client when trying to send the audit events.